### PR TITLE
Add arithmetic simplification

### PR DESCRIPTION
**Pull request by Markakd**
Do the following arithmetic simplification:

- Multiplication
- Division
- Mod
 
Currently doesn't support simplifying signed type division and remainder.

### CODE REVIEW

File: angr/analyses/decompiler/optimization_passes/multi_simplifier.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+            old_block = None
+
+            while new_block != old_block:
+                old_block = new_block
+                new_block = self.engine.process(state=self.state.copy(), block=old_block.copy())
+                _l.debug("new block: %s"%(new_block.statements))
+
+            self._update_block(block, new_block)
+
+AnalysesHub.register_default("MultiSimplifier", MultiSimplifier)
```
Comment (ltfish) [inline angr/analyses/decompiler/optimization_passes/multi_simplifier.py]: New line at the end of the file.

File: angr/analyses/decompiler/optimization_passes/mod_simplifier.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            old_block = None
+
+            while new_block != old_block:
+                old_block = new_block
+                new_block = self.engine.process(state=self.state.copy(), block=old_block.copy())
+                _l.debug("new block: %s"%(new_block.statements))
+
+            self._update_block(block, new_block)
+
+AnalysesHub.register_default("ModSimplifier", ModSimplifier)
```
Comment (ltfish) [inline angr/analyses/decompiler/optimization_passes/mod_simplifier.py]: New line at the end of the file.

File: angr/analyses/decompiler/optimization_passes/engine_base.py [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+                return Expr.Convert(expr.idx, operand_expr.from_bits, expr.to_bits, expr.is_signed, 
+                                    operand_expr.operand, **expr.tags)
+        elif type(operand_expr) is Expr.Const:
+            # do the conversion right away
+            value = operand_expr.value
+            mask = (2 ** expr.to_bits) - 1
+            value &= mask
+            return Expr.Const(expr.idx, operand_expr.variable, value, expr.to_bits, **expr.tags)
+        elif type(operand_expr) is Expr.BinaryOp \
+            and operand_expr.op in ['Mul', 'Shl', 'Div', 'DivMod', 'Add', 'Sub']:
```
Comment (ltfish) [inline angr/analyses/decompiler/optimization_passes/engine_base.py]: Make the list a set for better performance.

**Comment by ltfish** [comment]
Thanks for the PR! I'll hit merge once my nitpicks are addressed.

**Comment by Markakd** [comment]
Done. Anything else need to be changed?

**Comment by ltfish** [comment]
Can you please fix the linting errors that pylint reports? It's reported in the CI. Thanks.

**Comment by ltfish** [comment]
Merged. Thanks!

**Pull request merged**